dotnet/src/Connectors/Connectors.Memory.Pinecone/PineconeVectorStoreRecordCollection.cs

```diff
@@ -90,61 +79,112 @@ public PineconeVectorStoreRecordCollection(Sdk.PineconeClient pineconeClient, st
     }
 
     /// <inheritdoc />
-    public virtual async Task<bool> CollectionExistsAsync(CancellationToken cancellationToken = default)
-    {
-        var result = await this.RunOperationAsync(
-            CollectionExistsName,
+    public virtual Task<bool> CollectionExistsAsync(CancellationToken cancellationToken = default)
+        => this.RunCollectionOperationAsync(
+            "CollectionExists",
             async () =>
             {
-                var collections = await this._pineconeClient.ListIndexes(cancellationToken).ConfigureAwait(false);
+                var collections = await this._pineconeClient.ListIndexesAsync(cancellationToken: cancellationToken).ConfigureAwait(false);
 
-                return collections.Any(x => x.Name == this.CollectionName);
-            }).ConfigureAwait(false);
-
-        return result;
-    }
+                return collections.Indexes?.Any(x => x.Name == this.CollectionName) is true;
+            });
 
     /// <inheritdoc />
-    public virtual async Task CreateCollectionAsync(CancellationToken cancellationToken = default)
+    public virtual Task CreateCollectionAsync(CancellationToken cancellationToken = default)
     {
         // we already run through record property validation, so a single VectorStoreRecordVectorProperty is guaranteed.
         var vectorProperty = this._propertyReader.VectorProperty!;
-        var (dimension, metric) = PineconeVectorStoreCollectionCreateMapping.MapServerlessIndex(vectorProperty);
-
-        await this.RunOperationAsync(
-            CreateCollectionName,
-            () => this._pineconeClient.CreateServerlessIndex(
-                this.CollectionName,
-                dimension,
-                metric,
-                this._options.ServerlessIndexCloud,
-                this._options.ServerlessIndexRegion,
-                cancellationToken)).ConfigureAwait(false);
+
+        if (!string.IsNullOrEmpty(vectorProperty.IndexKind))
+        {
+            throw new InvalidOperationException(
+                $"'{vectorProperty.IndexKind}' is not supported. Pinecone supports only Pinecone Graph Algorithm (PGA), which is always enabled.");
+        }
```

Comment:
We should also allow the string `PGA`, since we are saying that we support that (also see https://learn.microsoft.com/en-us/semantic-kernel/concepts/vector-store-connectors/out-of-the-box-connectors/pinecone-connector).

```suggestion
        if (!string.IsNullOrEmpty(vectorProperty.IndexKind) && !string.CompareOrdinal(vectorProperty.IndexKind, "PGA"))
        {
            throw new InvalidOperationException(
                $"IndexKind of '{vectorProperty.IndexKind}' for property '{vectorProperty.PropertyName}' is not supported. Pinecone only supports 'PGA' (Pinecone Graph Algorithm), which is always enabled.");
        }
```